v3/docs/gs/sysreqs.md

```diff
@@ -8,18 +8,25 @@ The following are system requirements for running HSE applications.
 Hardware requirements are largely dictated by the application embedding HSE
 and the amount of data stored.  The following are general guidelines.
 
-* **Architecture**: 64-bit Intel&reg; and AMD (x86_64); 64-bit IBM Z&reg; (s390x)
+* **Architecture**: 64-bit Intel&reg;/AMD (x86_64) or IBM Z&reg; (s390x)
 * **Memory**: 32 GB or more
 * **Block Storage**: SSD volumes *only*; use NVMe for best performance
 * **Persistent Memory** (optional): must support a DAX-enabled file system
 
-If a KVDB [media class](storage.md#media-classes) is configured on multiple
-block storage devices, such
-as when using XFS with LVM, performance can be **significantly** improved by
-balancing these devices across NUMA nodes.
-Tools like [`lstopo`](https://linux.die.net/man/1/lstopo) can
-be helpful in creating and verifying a balanced configuration.
+If a file system storing a KVDB [media class](storage.md#media-classes) is configured on a
+logical block device comprising multiple physical block devices, such as when using XFS with LVM,
+performance can be **significantly** improved by balancing these physical devices across NUMA nodes.
+Tools like [`lstopo(1)`](https://linux.die.net/man/1/lstopo) can be helpful in creating and
+verifying a balanced configuration.
 
+For the single logical or physical block device configured with a file system for use with HSE,
+performance can also be **significantly** improved by setting the read-ahead to the greater
+of the device default or 128 KiB.  For example:
+
+```shell
+cat /sys/block/nvme0n1/queue/read_ahead_kb
+128
+```
```

Comment:
There are many performance tuning tips that could be documented.  Is this one so important it should be called out here while others are ignored?

Reply:
I think this one is important because of our dependency on the page cache for our DB caching. The readahead is critical to maintain an optimal tree shape and for optimal cursor performance in mongo replication.

In the context of 3.0, it is important that we call it out as we hardcode the RA size to 128K. So, if an user has configured 'read_ahead_kb' to something lower than 128K, it would result in the kernel trimming our 128K read-ahead requests to what's configured in 'read_ahead_kb' thus reducing perf.

In 3.1, we will be smarter about this by querying the 'read_ahead_kb' from sysfs for each media class FS at open time, clamping it between [0, 1M] and using the clamped value versus the hardcoded 128K in 3.0.

Reply:
Should we create a section focused on performance tuning?  Or consolidate it into the existing "Best Practices" section?  Right now we have tuning info here and in the "Benchmarking" section:

https://hse-project.github.io/2.x/res/benchmarking/

We also have API performance tips under "Best Practices" and "Concepts":

https://hse-project.github.io/2.x/dev/bp/
https://hse-project.github.io/2.x/dev/concepts/

Reply:
Here's the logic behind the current organization:

- System Requirements focuses purely on HW/SW requirements for using HSE including recommended system SW configuration
- Best Practices is a subsection of Development HSE Apps and focuses purely on best practices for using the HSE API when developing HSE Apps
- Benchmarking Tips basically reminds people benching HSE to make sure they adhere to the System Requirments, including configuring the system SW, and points out KVDB/KVS config parameters that are especially relevant for benchmarking.  I think it's important to call these things out in a section dedicated to benchmarking, given how important benchmark data can be in decision making.

We can discuss further, but so far there haven't been any external issues related to the current organization.